src/xmipp/libraries/data/pdb.cpp

```diff
@@ -540,13 +551,20 @@ void PDBRichPhantom::write(const FileName &fnPDB)
     for (size_t i=0; i<imax; ++i)
     {
     	const RichAtom &atom=atomList[i];
-        char result[5+1];
-        const char* errmsg = hy36encode(5, (int)i, result);
-    	fprintf (fh_out,"ATOM  %5s %4s%c%-4s%c%4d%c   %8.3f%8.3f%8.3f%6.2f%6.2f      %4s\n",
-    			result,atom.name.c_str(),
-    			atom.altloc,atom.resname.c_str(),atom.chainid,
-    			atom.resseq,atom.icode,atom.x,atom.y,atom.z,atom.occupancy,atom.bfactor,
-    			atom.name.c_str());
+        char serial[5+1];
+        if (const char* errmsg3 = hy36encode(5, atom.serial, serial); errmsg3) {
```

Comment:
see my previous comment
also I don't see the `i` used anywhere, even though it should be used as a position of the ATOM

Reply:
I don't think we should use `i` if there is already a value in the data. We probably need a check for this

Reply:
This is now included if we get an error using atom.serial